cmd/infra/azure/create.go

```diff
@@ -0,0 +1,338 @@
+	"github.com/Azure/azure-sdk-for-go/services/preview/authorization/mgmt/2020-04-01-preview/authorization"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2020-10-01/resources"
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2021-04-01/storage"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/hashicorp/go-uuid"
+
+	// This is the same client as terraform uses: https://github.com/hashicorp/terraform-provider-azurerm/blob/b0c897055329438be6a3a159f6ffac4e1ce958f2/internal/services/storage/blobs.go#L17
+	// The one from the azure sdk is cumbersome to use (distinct authorizer, requires to manually construct the full target url), and only allows upload from url for files that are not bigger than 256M.
+	"github.com/tombuildsstuff/giovanni/storage/2019-12-12/blob/blobs"
+)
+
+type CreateInfraOptions struct {
+	Name            string
+	Location        string
+	InfraID         string
+	CredentialsFile string
+	OutputFile      string
+}
+
+func NewCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "azure",
+		Short:        "Creates Azure infrastructure resources for a cluster",
+		SilenceUsage: true,
+	}
+
+	opts := CreateInfraOptions{
+		Location: "eastus",
+	}
+
+	cmd.Flags().StringVar(&opts.InfraID, "infra-id", opts.InfraID, "Cluster ID(required)")
+	cmd.Flags().StringVar(&opts.CredentialsFile, "azure-creds", opts.CredentialsFile, "Path to a credentials file (required)")
+	cmd.Flags().StringVar(&opts.Location, "location", opts.Location, "Location where cluster infra should be created")
+	cmd.Flags().StringVar(&opts.Name, "name", opts.Name, "A name for the cluster")
+	cmd.Flags().StringVar(&opts.OutputFile, "output-file", opts.OutputFile, "Path to file that will contain output information from infra resources (optional)")
+
+	cmd.MarkFlagRequired("infra-id")
+	cmd.MarkFlagRequired("azure-creds")
+	cmd.MarkFlagRequired("name")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if _, err := opts.Run(cmd.Context()); err != nil {
+			log.Log.Error(err, "Failed to create infrastructure")
+			return err
+		}
+		log.Log.Info("Successfully created infrastructure")
+		return nil
+	}
+
+	return cmd
+}
+
+func readCredentials(path string) (*apifixtures.AzureCreds, error) {
+	raw, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read from %s: %w", path, err)
+	}
+
+	var result apifixtures.AzureCreds
+	if err := yaml.Unmarshal(raw, &result); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal credentials: %w", err)
+	}
+
+	return &result, nil
+}
+
+type CreateInfraOutput struct {
+	Location          string `json:"region"`
+	ResourceGroupName string `json:"resourceGroupName"`
+	VNetID            string `json:"vnetID"`
+	VnetName          string `json:"vnetName"`
+	BootImageID       string `json:"bootImageID"`
```

Comment:
It is the disk image from which nodes boot. It is created once at infra install time and never updated, which means that in most cases it will be outdated which results in an ostree update during the nodes first boot. The approach is copied from how selfhosted OCP does it.